### ISSUE TITLE
Request: To toggle map view off so application can have very small footprint.

### ISSUE THREAD

**Issue by ukmerlin** [jaszhix/NoMansConnect#108]
Unless this is a feature request, ALL of the following is required:

- Description of the problem
- Windows version
- NMC version
- Absolute path to your storage_x.hg files.
- Absolute path to your NMS install directory.

Use Pastebin of directly attach these files (Users/<Username>/AppData/Roaming/NoMansConnect):

- NMC.log
- settings.json

- How can this bug be reproduced?

**Comment by jaszhix** [comment]
Its been a while since I've profiled the app, so this is worth looking into. When resetting remote locations, memory decreases by about 300-400MB for me. As long as the 3D map is disabled, CPU usage should be low.

This is really a bug, and I don't think working around a performance issue with another user option is a good fix. The memory issue is related to how many locations get loaded - the map components compounds the issue. This should get fixed in the map first.

We also should ask ourselves how much RAM is acceptable to throw at an auxiliary niche app for a game that doesn't normally run in the background. In my testing it generally consumes 1 GB. Assuming NMC only runs while NMS is running (which too will consume 1GB+), it seems like a reasonable footprint given the context of gaming, which machines used for that purpose typically have 8GB or more. 

If you are not running low on memory while playing the game, and not using the 3D map, NMC should have a very low impact on NMS' frame rate and responsiveness.

**Comment by ukmerlin** [comment]
I am sorry it was an unfortunate use of the word footprint. I am not really talking about memory but the screen space. I dont find the map at all useful and I usually have the app open with others like steam friends / chat so I was looking for a way to make it take less space on the desktop without losing any of the real information.

**Comment by jaszhix** [comment]
Oh, ok - in that case it could be accommodated.

**Comment by jaszhix** [comment]
Added an option to 1.8.0.

**Comment by ukmerlin** [comment]
Many thanks :)

**Comment by ukmerlin** [comment]
Sorry but tested this out and although the map disappears the overall window will not resize down anyway beyond where it would have if the map was still there.

**Comment by jaszhix** [comment]
I'm able to resize the window after disabling the map, have you tried unmaximizing?

**Comment by ukmerlin** [comment]
OK so there is only a maximize button, pressed it, filled whole screen (Ultrawide) Although there is no restore window I can grab the corner and bring it back. But it still only sizes down to the same size as it would with the map showing so there is a blank box in the middle.

**Comment by jaszhix** [comment]
This is what I'm able to arrive at with resizing the window:

![image](https://user-images.githubusercontent.com/6859057/63641909-bf0ba800-c67b-11e9-8f4b-bfe631333dd8.png)

I know there's some issues with window positioning when unmaximizing, but not understanding this particular issue.

**Comment by jaszhix** [comment]
Yeah, I guess you mean having the location list fill the remaining space while maximized. Since the columns are determined by the setting in the Registered Locations drop down menu, this doesn't fill automatically, but I could add options for more columns.

**Comment by ukmerlin** [comment]
Looks like my reply did not get thru. So from this view go ahead and go down to 1 column view.

**Comment by jaszhix** [comment]
OK, we'll reopen this and change the scope of the issue to make the app's layout more responsive to window size.
